Sources/SwiftParser/Statements.swift

```diff
@@ -877,11 +918,66 @@ extension Parser.Lookahead {
         // For example, could be the function call "discard()".
         return false
       }
+
+    case .then:
+      return isStartOfThenStatement(preferExpr: preferExpr)
+
     case nil:
       return false
     }
   }
 
+  /// Whether we're currently at a `then` token that should be parsed as a
+  /// `then` statement.
+  mutating func isStartOfThenStatement(preferExpr: Bool) -> Bool {
```

Comment:
`atStartOfThenStatement` would be more consistent with the other names (https://github.com/apple/swift-syntax/issues/1976). I also want to change the functions in `Lookahead` (https://github.com/apple/swift-syntax/issues/2001)

```suggestion
  mutating func atStartOfThenStatement(preferExpr: Bool) -> Bool {
```